backend/app/api/routes/documents.py

```diff
@@ -290,6 +311,23 @@ async def process_multiple_documents(
     return {"message": "Processing started for multiple files", "documents": results}
 
 
+@router.get("/by-course/{course_id}")
```

Comment:
```suggestion
@router.get("/{id}/documents")
```

Why not have it like this and maybe move it to the courses route?